### PR TITLE
Add test to check plugin has been attached to request obj

### LINKED ISSUE #8: Add test for new function of plugin
Create a test which check that the [function](https://github.com/SimonLab/tags-system/blob/5ddb845798225a4522d91409f457afbd3034583e/index.js#L3) has been added to the request object of the server.

To Do:
- [x] Install tape and istanbul
- [x] Write a test to find the ```typeof``` request.jackmisawesome is a function.

### PR DESCRIPTION
**Pull request by mk4111**
Closes #8 & #18

Adds test to check that:
- [x] we reach the endpoint in which we attach the plugin to the request obj
- [x] that the method exists and is a function
- [x] that the contents are what we expect it to be

### CODE REVIEW

**Comment by mk4111** [comment]
@SimonLab - could you please have a look at this PR when you get the time? Thank youuu! :D

File: setup-test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  }, function(result) {
+    t.equal(result.statusCode, 200, 'Looks for the endpoint where method is attached to request object.');
+    t.equal(typeof result.request.jackmisawesome, 'function', 'Checks to see if method exists and is a function');
+    t.equal(result.request.jackmisawesome(), 'Abracadabra!', 'Checks to see if method content is as expected');
+    t.end();
+  });
+});
+
+tape.onFinish(function(){
+  server.stop(function() {});
```
Comment (SimonLab) [inline setup-test.js]: I will prefer to pass server.stop in the t.end function. Like this each tests are independent and we avoid side effects on other tests (we might need to add the server inside the tape test

```
t.end(server.stop)
```

Comment (mk4111) [inline setup-test.js]: I think that looks _much_ better! I'll do exactly that :) Thank you!

Comment (mk4111) [inline setup-test.js]: @SimonLab - can I make a separate PR to fix #18?
